Universidade Federal do Rio Grande do Sul (UFRGS)   
Programa de Pós-Graduação em Engenharia Civil (PPGEC)   

# PEC00144: Experimental Methods in Civil Engineering


### Part I: Analysis
[2.   Dimensional analysis and similarity](#section_2)   
[2.1. Quantities, units, dimensions](#section_21)   
[2.2. The Theorem of Pi's by Vaschy-Buckinghan](#section_22)   
[2.3. Change of units and change of scales](#section_23)   
[2.4. Reduced scale models](#section_24)   

---
_Prof. Marcelo M. Rocha, Dr.techn._ [(ORCID)](https://orcid.org/0000-0001-5640-1020)  
_Porto Alegre, RS, Brazil_ 

## 2. Análise dimensional e similaridade <a name="section_2"></a> 

### 2.1. Grandezas, dimensões, unidades  <a name="section_21"></a> 

O Prof. G.I. Barenblatt escreveu em seu livro _Análise Dimensional_ (1987):   

>_... De fato, a ideia sobre a qual a análise dimensional se baseia é muito simples, e pode ser compreendida por todos:  leis físicas não dependem de arbitrariedades na escolha das unidades básicas de medida. Uma conclusão importante pode ser derivada desta ideia simples, usando argumentos simples: as funções que expressam leis físicas devem possuir uma certa propriedade fundamental, a qual em matemática é chamada de_ homogeneidade generalizada _ou simetria. Esta propriedade permite que o número de argumentos nestas  funções possa ser reduzido, tornando-os mais fáceis de serem obtidos (calculando-os ou determinando-os experimentalmente). Isso é, de fato, todo o conteúdo da análise dimensional $-$ nada mais existe além disso._

Em outras palavras, as unidades que resultam no dois lados de uma equação devem ser as mesmas. Embora simples, essa ideia permitiu a cientistas como Newton, Rayleigh, Maxwell, Reynolds, Kolmogorov, Cauchy, e muitos outros, a proposição de princípios físicos fundamentais. 

Para ilustrar a maneira como a lei de _homogeneidade dimensional_ pode ser aplicada, vamos imaginar o problema da frequência de vibração livre, $f_{\rm n}$, de um sistema massa-mola.

<img src="resources/figSDOF.jpg" alt="figSDOF" width="360px"/>

Um sistema massa-mola (não amortecido) tem como propriedades básicas a rigidez da mola, $k$, e a massa oscilante, $m$. Deve haver portanto uma relação entre as três grandezas, $f_{\rm n}$, $k$, e $m$. Observe que essa lista de parâmetros decorre de uma _hipótese de modelo_, que pode ou não estar correta. Vamos tentar encontrar essa relação a partir de uma análise das _unidades de medida_ dessas três grandezas:

\begin{align*}
[ f_{\rm n}] &= [L]^{0}[M]^{0}[T]^{-1} \\
[ k ]        &= [L]^{0}[M]^{1}[T]^{-2} \\
[ m ]        &= [L]^{0}[M]^{1}[T]^{0}
\end{align*}

onde os colchetes simbolizam "_unidade de_". Pode-se observar que as unidades foram escritas em termos das unidades de apenas três _grandezas fundamentais_: comprimento, $L$, massa, $M$, e tempo, $T$. A maior parte dos princípios utilizados na mecânica dos sólidos podem ser expressos apenas com essas três unidades. Eventualmente pode ser necessário utilizar também a temperatura, $\Theta$. Na sequência omitiremos os colchetes sempre que não houver ambiguidade.

As grandezas diretamente subordinadas às unidades fundamentais são denominadas _grandezas fundamentais_, que no caso são o comprimento, a massa, e o tempo. As demais unidades e grandezas são denominadas _unidades e grandezas derivadas_, como por exemplo a rigidez e a frequência. A subordinação das unidades derivadas às unidades fundamentais é expressa através de fórmulas dimensionais, como essas apresentadas acima. O expoente aplicado a cada unidade fundamental é denominado _dimensão_ ou _expoente de dimensão_.

No caso do _Sistema Internacional_, SI, utilizamos o metro (m), o kilograma (kg) e o segundo (s) como unidades das grandezas $L$, $M$, e $T$, respectivamente. Assim:

\begin{align*}
\left [ f_{\rm n} \right ] &= {\rm m}^{0}{\rm kg}^{0}{\rm s}^{-1} = 1/{\rm s} = {\rm Hz}\\
\left [ k         \right ] &= {\rm m}^{0}{\rm kg}^{1}{\rm s}^{-2} = {\rm kg}/{\rm s}^{2} = {\rm N/m} \\
\left [ m         \right ] &= {\rm kg}
\end{align*}

Nos cálculos de escala que faremos mais adiante será utilizado o conceito de _dictionary_ do Python. Assim, podemos criar uma relação das grandezas relevantes no contexto desta disciplina. Essa relação está incluída no módulo ```PEC00144.py```, armazenada na variável ```par_dict```:

In [7]:
from PEC00144 import *       # imports the parameters dictionary "par_dict" and other stuff

print (par_dict)


{'length': (1.0, 0.0, 0.0), 'mass': (0.0, 1.0, 0.0), 'time': (0.0, 0.0, 1.0), 'area': (2.0, 0.0, 0.0), 'volume': (3.0, 0.0, 0.0), 'S_A': (3.0, 0.0, 0.0), 'I_A': (4.0, 0.0, 0.0), 'mass_L': (1.0, 1.0, 0.0), 'mass_S': (2.0, 1.0, 0.0), 'density': (-3.0, 1.0, 0.0), 'I_M': (2.0, 1.0, 0.0), 'speed': (1.0, 0.0, -1.0), 'accel': (1.0, 0.0, -2.0), 'omega': (0.0, 0.0, 1.0), 'freq': (0.0, 0.0, 1.0), 'alpha': (0.0, 0.0, -2.0), 'force': (1.0, 1.0, -2.0), 'moment': (2.0, 1.0, -2.0), 'energy': (2.0, 1.0, -2.0), 'stiff': (0.0, 1.0, -2.0), 'stress': (-1.0, 1.0, -2.0), 'EI': (3.0, 1.0, -2.0), 'nu_dyn': (-1.0, 1.0, -1.0), 'nu_kyn': (2.0, 0.0, -1.0)}


As tuplas associadas a cada chave no dicionário são os expoentes das respectivas fórmulas dimensionais. Grandezas como ângulo ($\theta$), deformação ($\epsilon$), coeficiente de Poison ($\nu$), e razão de amortecimento crítico ($\zeta$) são adimensionais, e não é necessário incluí-las neste dicionário.

Para iterar sobre as entradas do dicionário basta fazer:

In [8]:
for ID, exponent in par_dict.items():
    
    print('{0:7} : [L] = {1:2.0f}, [M] = {2:2.0f}, [T] = {3:2.0f}'.format(ID, *exponent))


length  : [L] =  1, [M] =  0, [T] =  0
mass    : [L] =  0, [M] =  1, [T] =  0
time    : [L] =  0, [M] =  0, [T] =  1
area    : [L] =  2, [M] =  0, [T] =  0
volume  : [L] =  3, [M] =  0, [T] =  0
S_A     : [L] =  3, [M] =  0, [T] =  0
I_A     : [L] =  4, [M] =  0, [T] =  0
mass_L  : [L] =  1, [M] =  1, [T] =  0
mass_S  : [L] =  2, [M] =  1, [T] =  0
density : [L] = -3, [M] =  1, [T] =  0
I_M     : [L] =  2, [M] =  1, [T] =  0
speed   : [L] =  1, [M] =  0, [T] = -1
accel   : [L] =  1, [M] =  0, [T] = -2
omega   : [L] =  0, [M] =  0, [T] =  1
freq    : [L] =  0, [M] =  0, [T] =  1
alpha   : [L] =  0, [M] =  0, [T] = -2
force   : [L] =  1, [M] =  1, [T] = -2
moment  : [L] =  2, [M] =  1, [T] = -2
energy  : [L] =  2, [M] =  1, [T] = -2
stiff   : [L] =  0, [M] =  1, [T] = -2
stress  : [L] = -1, [M] =  1, [T] = -2
EI      : [L] =  3, [M] =  1, [T] = -2
nu_dyn  : [L] = -1, [M] =  1, [T] = -1
nu_kyn  : [L] =  2, [M] =  0, [T] = -1


Para acessar os expoentes associados a uma chave específica através do identificador (ID) faz-se:

In [9]:
print(par_dict['freq'])
print(par_dict['stiff'])
print(par_dict['mass'])


(0.0, 0.0, 1.0)
(0.0, 1.0, -2.0)
(0.0, 1.0, 0.0)


### 2.2. O Teorema dos Pi's de Vaschy-Buckinghan  <a name="section_22"></a> 

Observa-se agora que os expoentes nas fórmulas dimensionais (as dimensões) das três grandezas envolvidas no fenômeno de vibração livre do sistema massa mola são linearmente dependentes, ou seja, as unidades de uma das grandezas pode ser expressa através de uma relação entre as demais:

$$ \left [ f_{\rm n} \right ]  = \left [ k \right ] ^{1/2} \left [ m \right ] ^{-1/2} $$

Essas relações podem a princípio não parecerem muito evidentes, mas veremos adiante que há procedimentos matemáticos para procurá-las. O importante é que a existência de uma relação implica na possibilidade de se definir uma nova grandeza _adimensional_:

$$ \Pi = \frac{\sqrt{k / m}}{f_{\rm n}} $$

onde o número $\Pi$, por ser adimensional, não será afetado por uma mudança no sistema de unidades. Ou seja, $\Pi$ pode ser investigado experimentalmente, com boas chances de ser uma constante ou uma função que independe da escala do experimento. Essa é a abordagem básica da análise dimensional, que permitiu que grandes cientistas apresentassem ao mundo números adimensionais que regem fenômenos importantes, tais como o número de Reynolds (forças de arrasto em corpos imersos em fluidos), de Froude (efeito da gravidade sobre o movimento), ou de Strouhal (frequência de desprendimento de vórtices alternados). Observa-se que não foram utilizados quaisquer outros princípios físicos para se chegar à equação acima, ou seja, a conclusão é baseada completamente nas dimensões das grandezas envolvidas.

De fato, a partir da equação diferencial de equilíbrio dinâmico de um sistema massa mola pode-se demonstrar que $\Pi = 2\pi$, ou seja, uma constante. O mesmo poderia ter sido descoberto experimentalmente,
permitindo que finalmente se chegasse a:

$$ f_{\rm n} = \frac{1}{2\pi} \sqrt{\frac{k}{m}} $$



### 2.3. Mudança de unidades de base e mudança de escala  <a name="section_23"></a> 

As fórmulas dimensionais permitem recalcular as unidades derivadas a partir de uma nova escolha das unidades de base. Esse cálculo pode ser usado para diversas finalidades, como por exemplo: (1) mudanças no sistema de unidades, como por exemplo do SI para o sistema britânico, ou (2) mudanças de escala das grandezas, para projeto de modelos reduzidos.

Uma vez identificados os parâmetros que figuram num dado processo físico, é possível formar uma _matriz dimensional_, na qual colunas são os três expoentes (ou dimensões) das grandezas fundamentais compondo as unidades de todos os parâmetros. No módulo ```PEC00144.py``` essa matriz é lida diretamente do dicionário ```par_dict```, e está disponível na variável ```dim_mat```, contendo _todas_ as grandezas disponíveis:

In [10]:
print(dim_mat)

[[ 1.  0.  0.]
 [ 0.  1.  0.]
 [ 0.  0.  1.]
 [ 2.  0.  0.]
 [ 3.  0.  0.]
 [ 3.  0.  0.]
 [ 4.  0.  0.]
 [ 1.  1.  0.]
 [ 2.  1.  0.]
 [-3.  1.  0.]
 [ 2.  1.  0.]
 [ 1.  0. -1.]
 [ 1.  0. -2.]
 [ 0.  0.  1.]
 [ 0.  0.  1.]
 [ 0.  0. -2.]
 [ 1.  1. -2.]
 [ 2.  1. -2.]
 [ 2.  1. -2.]
 [ 0.  1. -2.]
 [-1.  1. -2.]
 [ 3.  1. -2.]
 [-1.  1. -1.]
 [ 2.  0. -1.]]


A matriz dimensional apenas para as grandezas fundamentais pode ser obtida lendo-se os respectivos expoentes no dicionário ```par_dict```. Isso é feito com a função ```dim_base```, que tem como argumento as chaves das três grandezas:

In [11]:
DB =  dim_base(('length', 'mass', 'time'));   print(DB)

[[1. 0. 0.]
 [0. 1. 0.]
 [0. 0. 1.]]


A escolha das grandezas $L$, $M$, e $T$ como grandezas de base é de certa forma arbitrária, embora esteja bem estabelecida internacionalmente. Contudo, é possível substituir essas três grandezas e logo obter uma nova matriz dimensional para outras três grandezas derivadas.

Supondo o estudo de um fenômeno qualquer na base $LMT$, no qual se deseje utilizar uma nova base composta pelas grandezas $A$, $B$ e $C$. A matriz dimensional da nova base é: 

\begin{align*}
 A &= L^{a_{L}} M^{a_{M}} T^{a_{T}} \\
 B &= L^{b_{L}} M^{b_{M}} T^{b_{T}} \\
 C &= L^{c_{L}} M^{c_{M}} T^{c_{T}}
\end{align*}

Por exemplo, chamando a função  ```dim_base``` para três novas grandezas de base, especificada como sendo formada por $A = L$ (comprimento),  $B = \mu$ (massa específica), e $C = g$ (aceleração da gravidade):

In [12]:
DB =  dim_base(('length', 'density', 'accel'));   print(DB)

[[ 1.  0.  0.]
 [-3.  1.  0.]
 [ 1.  0. -2.]]


Tomando-se o logaritmo de todos os termos, tem-se:

\begin{align*}
\log A &= \log \left ( L^{a_{L}} M^{a_{M}} T^{a_{T}} \right ) = a_{L}\log L + a_{M}\log M + a_{T}\log T \\
\log B &= \log \left ( L^{b_{L}} M^{b_{M}} T^{b_{T}} \right ) = b_{L}\log L + b_{M}\log M + b_{T}\log T \\
\log C &= \log \left ( L^{c_{L}} M^{c_{M}} T^{c_{T}} \right ) = c_{L}\log L + c_{M}\log M + c_{T}\log T
\end{align*}

Na forma matricial:
\begin{equation}
\left [ \begin{array}{ccc}
         a_{L}& a_{M}& a_{T} \\
         b_{L}& b_{M}& b_{T} \\
         c_{L}& c_{M}& c_{T}
        \end{array} \right ]
\left [ \begin{array}{ccc}
        \log L \\
        \log M \\
        \log T
        \end{array} \right ] =
\left [ \begin{array}{ccc}
        \log A \\
        \log B \\
        \log C
        \end{array} \right ]
\end{equation}

Onde se observa que a matriz do lado esquerdo da equação é a matriz dimensional da nova base. Isolando-se o vetor com as grandezas de base têm-se:

\begin{equation}
\left [ \begin{array}{ccc}
        \log L \\
        \log M \\
        \log T
        \end{array} \right ] =
\left [ \begin{array}{ccc}
         a_{A}& a_{B}& a_{C} \\
         b_{A}& b_{B}& b_{C} \\
         c_{A}& c_{B}& c_{C}
        \end{array} \right ]
\left [ \begin{array}{ccc}
        \log A \\
        \log B \\
        \log C
        \end{array} \right ]
\end{equation}

onde:

\begin{equation}
\left [ \begin{array}{ccc}
      a_{A}& a_{B}& a_{C} \\
      b_{A}& b_{B}& b_{C} \\
      c_{A}& c_{B}& c_{C}
     \end{array} \right ] =
\left [ \begin{array}{ccc}
      a_{L}& a_{M}& a_{T} \\
      b_{L}& b_{M}& b_{T} \\
      c_{L}& c_{M}& c_{T}
     \end{array} \right ] ^{-1}
\end{equation}

Voltando-se agora à forma polinomial, chega-se a:

\begin{align*}
 L &= A^{a_{A}} B^{a_{B}} C^{a_{C}} \\
 M &= A^{b_{A}} B^{b_{B}} C^{b_{C}} \\
 T &= A^{c_{A}} B^{c_{B}} C^{c_{C}}
\end{align*}

Observa-se, portanto, que para se obter a matriz dimensional das grandezas fundamentais em uma nova base basta inverter a matriz dimensional desta nova base. Isso só é possível se essa matriz tem determinante positivo, ou seja, as grandezas que constituem a nova base devem ter expoentes de dimensão linearmente independentes entre si. Isso significa, por exemplo, que não se poderia escolher como novas grandezas de base a _velocidade_ (m/s), o _comprimento_ (m), e o _tempo_ (s), pois as unidades de uma da três pode se representada por uma relação entre as outras duas. Logo, a matriz dimensional não poderia ser invertida.

A inversão da matriz dimensional da nova base é feita em Python pelo módulo ```numpy``` como:

In [13]:
IDB = np.linalg.inv(DB);   print(IDB)

[[ 1.  -0.  -0. ]
 [ 3.   1.   0. ]
 [ 0.5 -0.  -0.5]]


Vejamos agora o que acontece com uma grandeza derivada, $X$, para a qual se deseje conhecer as dimensões na nova base. As dimensões na base $LMT$ são dadas por:

\begin{equation}
 X = L^{\alpha_{L}} M^{\beta_{M}} T^{\gamma_{T}}
\end{equation} 

e portanto:

\begin{equation}
 X = \left ( A^{a_{A}} B^{a_{B}} C^{a_{C}} \right )^{\alpha_{L}}
     \left ( A^{b_{A}} B^{b_{B}} C^{b_{C}} \right )^{\beta_{M}}
     \left ( A^{c_{A}} B^{c_{B}} C^{c_{C}} \right )^{\gamma_{T}} 
\end{equation}

Rearranjando-se os termos:

\begin{equation}
 X = A^{(a_{A}\alpha_{L} + b_{A}\beta_{M} + c_{A}\gamma_{T})}
     B^{(a_{B}\alpha_{L} + b_{B}\beta_{M} + c_{B}\gamma_{T})}
     C^{(a_{C}\alpha_{L} + b_{C}\beta_{M} + c_{C}\gamma_{T})}
   = A^{\alpha_{A}} B^{\beta_{B}} C^{\gamma_{C}}
\end{equation}

Observa-se que as dimensões na nova base podem ser calculadas com uma multiplicação matricial:

\begin{equation}
\left [ \begin{array}{ccc}
        \alpha_{A} \\
        \beta_{B} \\
        \gamma_{C}
        \end{array} \right ]  =
\left [ \begin{array}{ccc}
         a_{A}& a_{B}& a_{C} \\
         b_{A}& b_{B}& b_{C} \\
         c_{A}& c_{B}& c_{C}
        \end{array} \right ] ^{T} 
\left [ \begin{array}{ccc}
        \alpha_{L} \\
        \beta_{M} \\
        \gamma_{T}
        \end{array} \right ] 
\end{equation}

Com o módulo ```numpy``` essa multiplicação é feita da seguinte forma:

In [14]:
new_dim_mat = np.dot(IDB.T, dim_mat.T).T;   print(new_dim_mat)

[[ 1.   0.   0. ]
 [ 3.   1.   0. ]
 [ 0.5  0.  -0.5]
 [ 2.   0.   0. ]
 [ 3.   0.   0. ]
 [ 3.   0.   0. ]
 [ 4.   0.   0. ]
 [ 4.   1.   0. ]
 [ 5.   1.   0. ]
 [ 0.   1.   0. ]
 [ 5.   1.   0. ]
 [ 0.5  0.   0.5]
 [ 0.   0.   1. ]
 [ 0.5  0.  -0.5]
 [ 0.5  0.  -0.5]
 [-1.   0.   1. ]
 [ 3.   1.   1. ]
 [ 4.   1.   1. ]
 [ 4.   1.   1. ]
 [ 2.   1.   1. ]
 [ 1.   1.   1. ]
 [ 5.   1.   1. ]
 [ 1.5  1.   0.5]
 [ 1.5  0.   0.5]]


No código acima, o cálculo dos novos expoentes foi feito diretamente para TODAS as grandezas disponíveis no dicionário ```par_dict```.

Como já mencionado, uma das aplicações da mudançaa de base é a mudança de unidades das grandezas fundamentais. A nova matriz dimensional nos permite calcular o efeito desta mudança nas grandezas derivadas. Por exemplo, vamos admitir que desejamos utilizar a milha nautica ao invés do metro como unidade de comprimento, e a hora ao invés do segundo como unidade do tempo. Vejamos como isso repercute na unidade de velocidade, sendo que a fórmula de cálculo é a mesma que foi anteriormente utilizada para a dimensão:

\begin{equation}
 X = A^{\alpha_{A}} B^{\beta_{B}} C^{\gamma_{C}}
\end{equation}

No módulo ```PEC00144.py``` todo o procedimento descrito acima está automatizado na função ```new_scale```, demonstrado abaixo:

In [19]:
L_length = 1/1852      # 1 nautical mile are 1852 meters
L_mass   = 1/1         # 1 kilogram will remain kilogram
L_time   = 1/3600      # 1 hour is 3600 seconds

L_speed  = new_scale('speed', ('length', 'mass', 'time'), (L_length, L_mass, L_time))

print(L_speed)

1.9438444924406049


ou seja, a nova unidade da velocidade é o nó (_1 knot = 1 mile/hour_) sendo que 1m/s (sistema SI) corresponde a aproximadamente 1.94 nós.

Outro aspecto interessante desse cálculo é que exatamente o mesmo procedimento pode ser utilizado para se modificar a escala de alguns parâmetros que descrevem um fenômeno físico, e conhecer a repercussão desta mudança em outros parâmetros relevantes. Esta é a ferramenta básica utilizada no projeto de modelos reduzidos, que será discutido na próxima seção.

Por exemplo, abaixo está o cálculo da nova escala de velocidade para uma mudança na escala de comprimento, $L$,  mantendo-se a escala de massa específica, $\mu$, e da aceleração da gravidade, $g$ inalteradas:

In [18]:
L_length = 1/200      # escala de comprimento do modelo reduzido
L_dens   = 1/1        # density unchanged
L_accel  = 1/1        # acceleration unchanged

L_speed  = new_scale('speed', ('length', 'density', 'accel'), (L_length, L_dens, L_accel))

print(1/L_speed)

14.142135623730951


Ou seja, uma velocidade de 1m/s no modelo reduzido corresponde aproximadamente a 14.1m/s na escala real (também chamada de escala do protótipo). 

### 2.4. Projeto de modelos reduzidos  <a name="section_24"></a> 

Como exemplo, vamos projetar um modelo reduzido de uma estrutura qualquer, para o
qual teremos uma escala de comprimento 1:50 (o modelo ser\ah\ 50 vezes menor que a
estrutura real) e uma escala de acelera\cao\ 1:1 (mesma acelera\cao\ da gravidade
estar\ah\ atuando tanto sobre o modelo como sobre a estrutura real.
Esse tipo de projeto \eh\ \uh til quando desejamos conhecer os deslocamentos e as
deforma��es da estrutura sujeita ao peso pr\oh prio.

Adicionalmente, vamos estabelecer que o modelo ser\ah\ constru\ih do em alum\ih nio,
enquanto a estrutura real \eh\ feita de concreto.
Isso resulta numa escala para o m\oh dulo de elasticidade igual a 71GPa\,:\,205GPa, 
ou seja, 1:2.89.
A repercuss\ao\ dessa mudan\cc a de escala nas grandezas derivadas est\ah\ apresentada
na \uh ltima coluna da Tab.~\ref{tabFroude}.
Vamos examinar cuidadosamente esses resultados.

Se o modelo \eh\ para ser constru\ih do e utilizado em experimentos que respeitem
plenamente a semelhan\cc a com a estrutura real \eh\ necess\ah rio que todos os 
demais par\an metros sejam respeitados, o que n\ao\ \eh\ em geral poss\ih vel.
Por exemplo, observa-se que a massa espec\ih fica do modelo deve ser aproximadamente
16.5 vezes maior que a massa espec\ih fica da estrutura real.
Se isso n\ao\ for respeitado, os deslocamentos sob peso pr\oh prio n\ao\ respeitar\ah\ 
a escala de comprimento de 1:50, ficando aqu\eh m dos valores corretos, e o modelo 
ser\ah\ inadequado para os estudos.
Contudo, a massa espec\ih fica do alum\ih nio \eh\ quase tr\en s vezes {\em menor\/}
que a massa espec\ih fica do a\cc o, contrariando o requerido.
Isso significa que o modelo deveria receber acr\eh scimo de massa, por exemplo
gr\an nulos de chumbo, a qual deveria ser criteriosamente distribu\ih da e n\ao\ 
interferir na rigidez do modelo.
Percebe-se desde j\ah\ que projetar um modelo deste tipo n\ao\ \eh\ uma tarefa
trivial. 
